File: The JSON Format/Java/GUI/api-gui-app/src/main/java/com/dosomedev/view/MainWindow.java
package com.dosomedev.view;

import java.awt.GridBagConstraints;
import java.awt.GridBagLayout;
import java.awt.Insets;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

import javax.swing.JButton;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JTextField;

import com.dosomedev.data.ApiAccess;
import com.dosomedev.model.Animal;

public class MainWindow extends JPanel implements ActionListener {
    private JLabel lblIndex;
    private JTextField txtIndex;
    private JButton btnShow;

    private JLabel lblAnimal;
    private JTextField txtAnimal;

    private JLabel lblName;
    private JTextField txtName;

    private JLabel lblBreed;
    private JTextField txtBreed;

    private JLabel lblAge;
    private JTextField txtAge;

    public MainWindow() {
        this.initFields();
        this.setupLayout();
    }

    private void initFields() {
        // Initialize components.
        this.lblIndex = new JLabel("Index:");
        this.txtIndex = new JTextField(10);
        this.btnShow = new JButton("Show Animal Info");

        this.lblAnimal = new JLabel("Animal:");
        this.txtAnimal = new JTextField(20);
        this.txtAnimal.setEditable(false);

        this.lblName = new JLabel("Name:");
        this.txtName = new JTextField(20);
        this.txtName.setEditable(false);

        this.lblBreed = new JLabel("Breed");
        this.txtBreed = new JTextField(20);
        this.txtBreed.setEditable(false);

        this.lblAge = new JLabel("Age:");
        this.txtAge = new JTextField(20);
        this.txtAge.setEditable(false);
    }

    private void setupLayout() {
        // Define layout type.
        GridBagLayout layout = new GridBagLayout();
        setLayout(layout);
        GridBagConstraints constraints = new GridBagConstraints();

        // Define padding.
        constraints.insets = new Insets(5, 5, 5, 5);

        // Add components to panel.
        constraints.gridx = 0;
        constraints.gridy = 0;
        add(this.lblIndex, constraints);
        JPanel indexPanel = new JPanel();
        indexPanel.add(this.txtIndex);
        indexPanel.add(this.btnShow);
        constraints.gridx = 1;
        add(indexPanel, constraints);

        constraints.gridx = 0;
        constraints.gridy = 1;
        add(this.lblAnimal, constraints);
        constraints.gridx = 1;
        add(this.txtAnimal, constraints);

        constraints.gridx = 0;
        constraints.gridy = 2;
        add(this.lblName, constraints);
        constraints.gridx = 1;
        add(this.txtName, constraints);

        constraints.gridx = 0;
        constraints.gridy = 3;
        add(this.lblBreed, constraints);
        constraints.gridx = 1;
        add(this.txtBreed, constraints);

        constraints.gridx = 0;
        constraints.gridy = 4;
        add(this.lblAge, constraints);
        constraints.gridx = 1;
        add(this.txtAge, constraints);

        // Register action listener.
        this.btnShow.addActionListener(this);
    }

    private void setFields(Animal animal) {
        if (animal == null) {
            this.txtAnimal.setText("");
            this.txtName.setText("");
            this.txtBreed.setText("");
            this.txtAge.setText("");
        } else {
            this.txtAnimal.setText(animal.getAnimal());
            this.txtName.setText(animal.getName());
            this.txtBreed.setText(animal.getBreed());
            this.txtAge.setText(animal.getAge().toString());
        }
    }

    @Override
    public void actionPerformed(ActionEvent e) {
        // Capture button event.
        if (e.getSource() == this.btnShow) {
            // Check if the input is a valid number.
            try {
                // Get index input.
                int indexNumber = Integer.parseInt(this.txtIndex.getText());

                // Get animals from web service.
                ApiAccess apiAccess = ApiAccess.getInstance();
                Animal[] animals = apiAccess.getAnimals();

                // Check if index valid.
                if (animals != null && 0 <= indexNumber && indexNumber < animals.length) {
                    System.out.println("Setting fields.");
                    this.setFields(animals[indexNumber]);
                } else {
                    throw new IndexOutOfBoundsException("Invalid index. Please enter an existing index.");
                }
            } catch (NumberFormatException ex) {
                this.setFields(null);
                System.out.println("Invalid input. Please enter a number.");
            } catch (IndexOutOfBoundsException ex) {
                this.setFields(null);
                System.out.println(ex.getMessage());
            }
        }
    }
}

File: The JSON Format/Python/GUI/view/main_window.py
import tkinter as tk
from data.api_access import ApiAccess
from model.animal import Animal

class MainWindow:
    def __init__(self):
        # Create main window.
        self.window = tk.Tk()
        self.window.title("Animal Info")

        self.init_fields()
        self.setup_layout()
        
    def init_fields(self):
        # Create fields.
        self.lbl_index = tk.Label(self.window, text="Index:")
        self.txt_index = tk.Entry(self.window, width=15)
        self.btn_show = tk.Button(self.window, text="Show Animal Info", command=self.btn_show_clicked)

        self.lbl_animal = tk.Label(self.window, text="Animal:")
        self.txt_animal = tk.Entry(self.window, width=30)

        self.lbl_name = tk.Label(self.window, text="Name:")
        self.txt_name = tk.Entry(self.window, width=30)

        self.lbl_breed = tk.Label(self.window, text="Breed:")
        self.txt_breed = tk.Entry(self.window, width=30)

        self.lbl_age = tk.Label(self.window, text="Age:")
        self.txt_age = tk.Entry(self.window, width=30)

        self.set_fields_readonly()

    def setup_layout(self):
        # Set layout.
        self.lbl_index.grid(row=0, column=0, padx=3, pady=3)
        self.txt_index.grid(row=0, column=1, padx=3, pady=3)
        self.btn_show.grid(row=0, column=2, padx=3, pady=3)

        self.lbl_animal.grid(row=1, column=0, padx=3, pady=3)
        self.txt_animal.grid(row=1, column=1, columnspan=2, padx=3, pady=3)

        self.lbl_name.grid(row=2, column=0, padx=3, pady=3)
        self.txt_name.grid(row=2, column=1, columnspan=2, padx=3, pady=3)

        self.lbl_breed.grid(row=3, column=0, padx=3, pady=3)
        self.txt_breed.grid(row=3, column=1, columnspan=2, padx=3, pady=3)

        self.lbl_age.grid(row=4, column=0, padx=3, pady=3)
        self.txt_age.grid(row=4, column=1, columnspan=2, padx=3, pady=3)

    def set_visible(self):
        self.window.mainloop()

    def set_fields_readonly(self):
        self.txt_animal.configure(state="readonly")
        self.txt_name.configure(state="readonly")
        self.txt_breed.configure(state="readonly")
        self.txt_age.configure(state="readonly")

    def set_fields_normal(self):
        self.txt_animal.configure(state="normal")
        self.txt_name.configure(state="normal")
        self.txt_breed.configure(state="normal")
        self.txt_age.configure(state="normal")

    def set_fields(self, animal: Animal):
        self.set_fields_normal()
        
        self.txt_animal.delete(0, tk.END)
        self.txt_name.delete(0, tk.END)
        self.txt_breed.delete(0, tk.END)
        self.txt_age.delete(0, tk.END)

        if animal is not None:
            self.txt_animal.insert(0, animal.animal_type)
            self.txt_name.insert(0, animal.name)
            self.txt_breed.insert(0, animal.breed)
            self.txt_age.insert(0, animal.age)

        self.set_fields_readonly()

    def btn_show_clicked(self):
        # Get index input.
        txt_index_value = self.txt_index.get()
        if txt_index_value.isdigit():
            # Convert index input to int.
            index_number = int(txt_index_value)

            # Get animals from web service.
            api_access = ApiAccess()
            animals = api_access.get_animals()

            # Check if index is valid.
            if 0 <= index_number < len(animals):
                print("Setting fields.")
                self.set_fields(animals[index_number])
            else:
                self.set_fields(None)
                print("Invalid index. Please enter an existing index.")

        else:
            self.set_fields(None)
            print("Invalid input. Please enter a number.")
